lib/matplotlib/texmanager.py

```diff
@@ -398,30 +399,34 @@ def make_dvi(self, tex, fontsize):
         if DEBUG or not os.path.exists(dvifile):
             texfile = self.make_tex(tex, fontsize)
             outfile = basefile + '.output'
-            command = self._get_shell_cmd(
-                'cd "%s"' % self.texcache,
-                'latex -interaction=nonstopmode %s > "%s"' %
-                (os.path.split(texfile)[-1], outfile))
+            command = [str("latex"), "-interaction=nonstopmode",
+                       os.path.split(texfile)[-1]]
             mpl.verbose.report(command, 'debug')
             with Locked(self.texcache):
-                exit_status = os.system(command)
-            try:
-                with open(outfile) as fh:
-                    report = fh.read()
-            except IOError:
-                report = 'No latex error report available.'
-            try:
-                os.stat(dvifile)
-                exists = True
-            except OSError:
-                exists = False
-            if exit_status or not exists:
-                raise RuntimeError(
-                    ('LaTeX was not able to process the following '
-                     'string:\n%s\nHere is the full report generated by '
-                     'LaTeX: \n\n' % repr(tex.encode('unicode_escape')) +
-                     report))
-            else:
+                try:
+                    with open(outfile, "w") as fout:
+                        subprocess.check_call(command,
+                                              cwd=self.texcache,
+                                              stdout=fout,
+                                              stderr=subprocess.STDOUT)
+                except subprocess.CalledProcessError as exc:
+                    try:
+                        with open(outfile) as fh:
+                            report = fh.read()
+                    except IOError:
+                        report = 'No latex error report available.'
+                    raise RuntimeError(
+                        ('LaTeX was not able to process the following '
+                         'string:\n%s\nLaTeX failed with: %s\n'
+                         'Here is the full report generated by LaTeX: '
+                         '\n\n' % (exc.output,
```

Comment:
I removed `exc.output`, but I don't think it is possible to switch to `check_output` (see above the other comment)